eng/Analyzers.targets

```diff
@@ -1,4 +1,8 @@
 <Project>
+  <PropertyGroup Condition="'$(UsingMicrosoftNoTargetsSdk)' == 'true' or '$(UsingMicrosoftDotNetSharedFrameworkSdk)' == 'true'">
+    <!-- Explicitly disable running analyzers to avoid trying to discover the correct ILLink tool pack for a project that has no sources. -->
+    <RunAnalyzers>false</RunAnalyzers>
+  </PropertyGroup>
```

Comment:
nit: Personally I would prefer this to be a condition directly on the property instead of defining a new group but that's just my own taste of style. We did the same for the DotNetBuildFromSource condition below...